apps/emqx/src/emqx_passwd.erl

```diff
@@ -139,11 +131,38 @@ compare_secure([X | RestX], [Y | RestY], Result) ->
 compare_secure([], [], Result) ->
     Result == 0.
 
+-define(MAX_DERIVED_KEY_LENGTH, (1 bsl 32 - 1)).
```

Comment:
copied from `pbkdf2`, we should limit it tow a much smaller value. 
should it be 64 (512 bit long)?

Reply:
It will be a breaking change if user is using very large DERIVED_KEY_LENGTH by configuration like `1 bsl 32 - 2`

Reply:
it will never return when it's too long.

Reply:
I will leave it for another PR